code/datums/quirks/negative.dm

```diff
@@ -455,6 +457,47 @@
 	to_chat(quirk_holder, "<span class='boldannounce'>Your [slot_string] has been replaced with a surplus prosthetic. It is fragile and will easily come apart under duress. Additionally, \
 	you need to use a welding tool and cables to repair it, instead of bruise packs and ointment.</span>")
 
+/datum/quirk/prosthetic_limbs
+	name = "Prosthetic Limbs"
+	desc = "Oops! All Prosthetics! Due to some truly cruel cosmic punishment, all your limbs have been taken from you."
+	icon = "tg-prosthetic-full"
+	value = -8
+	medical_record_text = "During physical examination, patient was found to have all prosthetic limbs"
+	hardcore_value = 6
+
+/datum/quirk/prosthetic_limbs/add_unique()
+	var/limb_slotLA = BODY_ZONE_L_ARM
+	var/limb_slotRA = BODY_ZONE_R_ARM
+	var/limb_slotLL = BODY_ZONE_L_LEG
+	var/limb_slotRL = BODY_ZONE_R_LEG
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/obj/item/bodypart/old_LA = human_holder.get_bodypart(limb_slotLA)
+	var/obj/item/bodypart/old_RA = human_holder.get_bodypart(limb_slotRA)
+	var/obj/item/bodypart/old_LL = human_holder.get_bodypart(limb_slotLL)
+	var/obj/item/bodypart/old_RL = human_holder.get_bodypart(limb_slotRL)
+
+	var/obj/item/bodypart/prostheticLA
+	var/obj/item/bodypart/prostheticRA
+	var/obj/item/bodypart/prostheticLL
+	var/obj/item/bodypart/prostheticRL
+
+	prostheticLA = new/obj/item/bodypart/l_arm/robot/surplus(quirk_holder)
+	prostheticRA = new/obj/item/bodypart/r_arm/robot/surplus(quirk_holder)
+	prostheticLL = new/obj/item/bodypart/l_leg/robot/surplus(quirk_holder)
+	prostheticRL = new/obj/item/bodypart/r_leg/robot/surplus(quirk_holder)
+
+	prostheticLA.replace_limb(human_holder)
+	prostheticRA.replace_limb(human_holder)
+	prostheticLL.replace_limb(human_holder)
+	prostheticRL.replace_limb(human_holder)
+	qdel(old_LA, old_RA, old_LL, old_RL)
```

Comment:
This is not ideal. Could you just make a list with the new types and use a for loop to replace the limbs? It’s also not necessary to store and delete all the old limbs, replace_limb probably has a paramater to just delete the replaced limb

Reply:
https://github.com/tgstation/tgstation/blob/master/code/modules/surgery/bodyparts/dismemberment.dm#L316 no parameter to delete the replaced limb, hence the existing quirk doing the same thing

Reply:
That code could also be better, but I think it’s a bit cruel to suggest an associative static list. I think doing what you suggested is good for now